transform/livereloadinject/livereloadinject.go

```diff
@@ -16,25 +16,41 @@ package livereloadinject
 import (
 	"bytes"
 	"fmt"
+	"net/url"
+	"strings"
 
 	"github.com/gohugoio/hugo/helpers"
 	"github.com/gohugoio/hugo/transform"
 )
 
 // New creates a function that can be used
 // to inject a script tag for the livereload JavaScript in a HTML document.
-func New(port int) transform.Transformer {
+func New(baseURL url.URL) transform.Transformer {
 	return func(ft transform.FromTo) error {
 		b := ft.From().Bytes()
 		endBodyTag := "</body>"
 		match := []byte(endBodyTag)
-		replaceTemplate := `<script data-no-instant>document.write('<script src="/livereload.js?port=%d&mindelay=10&v=2"></' + 'script>')</script>%s`
-		replace := []byte(fmt.Sprintf(replaceTemplate, port, endBodyTag))
+		if !strings.HasSuffix(baseURL.Path, "/") {
+			baseURL.Path += "/"
+		}
+
+		src := baseURL.Path + "livereload.js?mindelay=10&v=2"
+
+		src += "&port=" + baseURL.Port()
+
+		path := baseURL.Path + "livereload"
+		if strings.HasPrefix(path, "/") {
+			path = path[1:]
+		}
```

Comment:
given `strings.TrimPrefix(str, prefix)`  does exactly this internally, is there a particular motivation for handling this manually like this?

Reply:
No.